static/register.js

```diff
@@ -13,3 +13,9 @@ select.addEventListener('change', (event) => {
     form.style.display = "none";
   }
 })
+
+// For patientOfDoctor preventing refresh
+let form = document.getElementById("pre_form");
+
+function handleForm(event) { event.preventDefault(); }
+form.addEventListener('submit', handleForm);
```

Comment:
I think this would make sense to be in a different file specific to the patientOfDoctor page to prevent errors